### ISSUE TITLE
Extract UI components and make it reusable to other use cases

### ISSUE THREAD

**Issue by LintaoAmons** [Ramilito/kubectl.nvim#129]
I found this buffer UI and the way to interact with the UI is very desired for other scenarios like docker or any other CLI tools.

Hope there's standalone version of this ui components which can easily adapt to other use cases, and be used by others plugin authors

**Comment by Ramilito** [comment]
Interesting idea, haven't thought along those lines but that should be feasible.
The code already is a bit split up into different responsibilities and it should handle any json data.

**Comment by mosheavni** [comment]
i think it is possible but there are already plugins that provide some of the UI principles in this one.
as this is open source code, anyone can copy the logic freely, no @Ramilito ?

**Comment by Ramilito** [comment]
Yeah some of the UI things are already solved, what I took from this though is basically extract the table.lua and some of the buffer.lua stuff.
Being able to take json data and just get out a column based table basically.

**Comment by Ramilito** [comment]
This is a pretty big task with how tailored the code is to fit with kubernetes, I don't see this happening any time soon, let me know if anyone wants to try it out though.
